python/pyarrow/tests/test_compute.py

```diff
@@ -284,6 +284,66 @@ def test_take_indices_types():
             arr.take(indices)
 
 
+def test_take_on_chunked_array():
+    # ARROW-9504
+    arr = pa.chunked_array([
+        [
+            "m",
+            "J",
+            "q",
+            "k",
+            "t"
+        ],
+        [
+            "m",
+            "J",
+            "q",
+            "k",
+            "t"
+        ]
+    ])
+
+    indices = np.array([0, 5, 1, 6, 2, 7, 3, 8, 4, 9])
+    result = arr.take(indices)
+    expected = pa.chunked_array([
+        "m",
+        "m",
+        "J",
+        "J",
+        "q",
+        "q",
+        "k",
+        "k",
+        "t"
+        "t"
+    ])
```

Comment:
What's the point of a second example?

Reply:
To use a chunked array as indices because the number of output chunks depends on it.

Reply:
Second example?